utils/maps/map.go

```diff
@@ -0,0 +1,88 @@
+package maps
+
+import (
+	"strings"
+)
+
+const separator = "."
+
+// Map is a wrapper around maps[string]string that provides some helpers
+// above it that assume the maps is in the format that flatmap expects
+// (the result of Flatten).
+//
+// All modifying functions such as Delete are done in-place unless
+// otherwise noted.
+type Map map[string]string
+
+// Contains returns true if the maps contains the given key.
+func (m Map) Contains(key string) bool {
+	for _, k := range m.Keys() {
+		if k == key {
+			return true
+		}
+	}
+
+	return false
+}
+
+func (m Map) AsMap() map[string]string {
+	return m
+}
+
+// Delete deletes a key out of the maps with the given prefix.
+func (m Map) Delete(prefix string) {
+	for k := range m {
+		match := k == prefix
+		if !match {
+			if !strings.HasPrefix(k, prefix) {
```

Comment:
HasPrefix works on a sting if it is entirely prefix so you can remove the match I think

Reply:
I used the code as is from terraform and only made some small changes. I will make changes as part of further developments